chain/filecoin/account.go

```diff
@@ -29,10 +32,11 @@ func NewTxBuilder() TxBuilder {
 }
 
 // BuildTx receives transaction fields and constructs a new transaction.
-func (txBuilder TxBuilder) BuildTx(ctx context.Context, from, to address.Address, value, nonce, gasLimit, gasPrice, gasCap pack.U256, payload pack.Bytes) (account.Tx, error) {
-	filfrom, err := filaddress.NewFromString(string(from))
+func (txBuilder TxBuilder) BuildTx(ctx context.Context, fromPubKey *id.PubKey, to address.Address, value, nonce, gasLimit, gasPrice, gasCap pack.U256, payload pack.Bytes) (account.Tx, error) {
+	pubKeyUncompressed := eth_crypto.FromECDSAPub((*ecdsa.PublicKey)(fromPubKey))
+	filfrom, err := filaddress.NewSecp256k1Address(pubKeyUncompressed)
 	if err != nil {
-		return nil, fmt.Errorf("bad from address '%v': %v", from, err)
+		return nil, fmt.Errorf("bad from pubkey address '%v': %v", fromPubKey, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("bad from pubkey: %v", err)
```

Can remove this since it's no longer a readable address.